foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/AbstractPoetryMigration.java

```diff
@@ -0,0 +1,95 @@
+package com.boozallen.aissemble.upgrade.migration;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.boozallen.aissemble.upgrade.util.FileUtils;
+import com.boozallen.aissemble.upgrade.util.PythonPackageMigration;
+import com.electronwill.nightconfig.core.Config;
+import com.electronwill.nightconfig.core.file.FileConfig;
+
+/**
+ * 
+ * Baton migration used to migrate aiSSEMBLE python packages to the new naming convention and version in pyproject.toml files
+ */
+public abstract class AbstractPoetryMigration extends AbstractAissembleMigration {
+
+    private static final Logger logger = LoggerFactory.getLogger(AbstractPoetryMigration.class);
+    public static final String DEPENDENCIES_KEY = "tool.poetry.dependencies";
+
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        boolean shouldExecute = false;
+        FileConfig poetryConfig = FileConfig.of(file);
+        poetryConfig.load();
+        Optional<Config> dependenciesOpt = poetryConfig.getOptional(DEPENDENCIES_KEY);
+        if(dependenciesOpt.isPresent()) {
+            Config dependencies = dependenciesOpt.get();
+            shouldExecute = hasOldPackages(dependencies);
+        }
+        else {
+            logger.warn("Could not get dependencies for file {}", file.getAbsolutePath());
+        }
+        return shouldExecute;
+    }
+
+    private boolean hasOldPackages(Config currentPackages) {
+        boolean hasOldPackages = false;
+        for(PythonPackageMigration packageMigration : getPythonPackageMigrations()) {
+            if(currentPackages.contains(packageMigration.getOldName())) {
+                hasOldPackages = true;
+                break;
+            }
+        }
+        return hasOldPackages;
+    }
+
+    @Override
+    protected boolean performMigration(File file) {
+        boolean performedSuccessfully = false;
+        FileConfig pyproject = FileConfig.of(file);
+        pyproject.load();
+        Config dependencies = pyproject.get(DEPENDENCIES_KEY);
+        try {
+            performedSuccessfully = migrateOldPythonPackages(dependencies, file);
+        } catch (IOException e) {
+            logger.error(e.getMessage());
```

Comment:
S: if we get and exception, we should fail hard IMO.  Wrap in a `BatonException` and re-throw. 

```suggestion
            throw new BatonException("Could not migrate file: " + file.getPath(), e);
```

Reply:
Throws a BatonException.